src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -819,5 +819,30 @@ void M()
             Assert.Equal(@null, rightInfo.Nullability);
             Assert.Equal(notNull, rightInfo.ConvertedNullability);
         }
+
+        [Fact]
+        public void InferredDeclarationType()
```

Comment:
>InferredDeclarationType [](start = 20, length = 23)

Could you also test other kinds of implicit declarations?
`M(out var x);`
`var (x, y) = p;`
`foreach (var (x, y) in pCollection) ...`
`using (var x = d) ...`
`using var x = d;` (probably already covered by your change)

Reply:
Deconstruction and foreach deconstruction won't work yet (see #35010 which I was working on when I discovered this :)) 

I'll try the others and see how they fair.